BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/DiagnosticsModule/FileDiagnosticsSender.cs

```diff
@@ -0,0 +1,168 @@
+                            }
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        // no op
+                    }
+                }
+            }
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)
+            {
+                if (disposing)
+                {
+                    this.defaultTraceListener.Dispose();
+                }
+
+                this.disposedValue = true;
+            }
+        }
+
+        private static void WriteFileHeader(string logFilePath)
+        {
+            string[] lines =
+            {
+                // this.SelfDiagnosticsConfig,
+                ".NET SDK version: " + SdkVersionUtils.GetSdkVersion(string.Empty),
+                string.Empty,
+            };
+
+            System.IO.File.WriteAllLines(logFilePath, lines);
+        }
+
+        private bool SetAndValidateLogsFolder(string fileDirectory, string fileName)
+        {
+            bool result = false;
+            try
+            {
+                if (!string.IsNullOrWhiteSpace(fileDirectory) && !string.IsNullOrWhiteSpace(fileName))
+                {
+                    // Validate
+                    string expandedDirectory = Environment.ExpandEnvironmentVariables(fileDirectory);
+                    var logsDirectory = new DirectoryInfo(expandedDirectory);
+                    FileHelper.TestDirectoryPermissions(logsDirectory);
+
+                    string fullLogFileName = Path.Combine(expandedDirectory, fileName);
+
+                    // Set
+                    this.LogFilePath = fullLogFileName;
+                    WriteFileHeader(fullLogFileName);
+
+                    result = true;
+                }
+            }
+            catch (Exception ex)
+            {
+                // NotSupportedException: The given path's format is not supported
+                // UnauthorizedAccessException
+                // ArgumentException: // Path does not specify a valid file path or contains invalid DirectoryInfo characters.
+                // DirectoryNotFoundException: The specified path is invalid, such as being on an unmapped drive.
+                // IOException: The subdirectory cannot be created. -or- A file or directory already has the name specified by path. -or-  The specified path, file name, or both exceed the system-defined maximum length.
+                // SecurityException: The caller does not have code access permission to create the directory.
+
+                Trace.WriteLine(new Exception(Invariant($"{nameof(FileDiagnosticsSender)}.{nameof(this.SetAndValidateLogsFolder)} failed. Path: {this.logDirectory} File: {this.logFileName}; Error: {ex.Message}{Environment.NewLine}")));
```

Comment:
What does Trace.WriteLine do? Are we using this anywhere else/

Reply:
I saw Kennedy mention it on Sean's PR and I haven't been able to stop thinking about it.
If something failed in here, we wouldn't be able to know.
I can remove it, what do you think?

To listen to Trace messages, you would have to modify the app.config to listen to these events.

We don't do this anywhere else. The rest of the product logs to ETW/DiagnosticsTelemetryModule.

Reply:
Our logging mechanism is EventSource always - so I'll just use it and not introduce anything new.

Reply:
removed